packages/core/src/StateMachine.ts

```diff
@@ -65,6 +65,7 @@ export class StateMachine<
   TTag extends string,
   TInput,
   TOutput,
+  TEmitted extends EventObject,
```

Comment:
we might need to default those since `@xstate/react` and other packages might depend on the number of arguments - we already had small issues like this but we can take another look at the fixing PRs to understand if the nature of both is the same